go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/cert-manager/cert-manager
 
-go 1.20
+go 1.21
```

Comment:
I suppose this changed because the 0.29 K8S libraries use this minimum Go version.
I'll add a note about that to the release notes when we do a release, for the benefit of anyone importing the 1.12 cert-manager module.
 * https://github.com/kubernetes/client-go/compare/v0.27.2...v0.29.7#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6